igcollect/statuscode2gaphite.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import datetime
+import time
+import socket
+
+from collections import Counter
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description='Script to collect statuscodes from nginx')
+    parser.add_argument('logfile',
+                        help='path to logfile')
+    parser.add_argument('-s', '--software', default='nginx',
+                        help='which tool is using it ')
+    parser.add_argument('-t', '--time', type=int, default=3,
+                        help='where to search for the time in the log file ')
+    parser.add_argument('-x', '--position', type=int, default=8,
+                        help='where to search for the code in the log file ')
+    parser.add_argument('-p', '--prefix',
+                        default='servers.{}.software.{}'.format(
+                            socket.gethostname().replace('.', '_'),
+                            parser.parse_args().software),
+                        help='the path to the value in Graphite ')
+    return parser.parse_args()
+
+
+def get_position(position, entry):
+    return entry[position]
+
+
+def get_lines(path):
+    log_file = open(path)
+    log = []
+    for line in log_file:
+        log.append(line.split(' '))
+    return log
```

Comment:
This loads the whole log to memory. Can't it be done using streams?

Reply:
this is already streaming :D

Reply:
No, it's not. You're allocating an array and pushong the whole file to it, line by line. If the file is a few hundred MiB, this script will consume this much memory.

Reply:
https://docs.python.org/3.5/library/io.html

Reply:
The problem at the end is to convert the text to a set of lists ^^

Reply:
You could do it in a streaming way, count codes as you go through the log file line by line.

Reply:
even on a forge server thi script uses a max of 1.1mb ram / run

Reply:
I would prefer to not take any chances here. This script is potentially disasterous.